### PR TITLE
Add Weather module

### PR DESCRIPTION
**Pull request by mdhanif-simformsolutions**
Jira ticket :- 
- https://timelessspace.atlassian.net/browse/NF-1256

### CODE REVIEW

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/PollBubble.swift [hunk 1]
```diff
@@ -131,6 +131,7 @@ public class PollBubble: UITableViewCell {
     }
 
     func configData(isSender: Bool) {
+        debugPrint("## Poll details :- \(content?.localState)")
```
Comment (parth-simformsolutions) [inline Sources/StreamChatUI/ChatMessageList/ChatMessage/PollBubble.swift]: @mdhanif-simformsolutions  remove debug print.

Comment (mdhanif-simformsolutions) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/PollBubble.swift]: Updated this 
thanks .

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/WeatherCell.swift [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+        }
+    }
+
+    func configureCell(isSender: Bool) {
+
+        isCurrentMessageSend = !(content?.localState == .pendingSend || content?.localState == .sending || content?.type == .ephemeral)
+
+        guard let content = content?.extraData,
+              let weather = getExtraData(key: "weather"),
+              let currentLocation = weather["currentLocation"],
```
Comment (ajay-simform) [inline Sources/StreamChatUI/ChatMessageList/ChatMessage/WeatherCell.swift]: @mdhanif-simformsolutions Please move these variable to extraData helper

File: Sources/StreamChatUI/Composer/ComposerVC.swift [hunk 3]
```diff
@@ -339,6 +346,11 @@ open class ComposerVC: _ViewController,
         }
     }
     var shouldToggleEmojiButton = true
+    private var locationManager: CLLocationManager = {
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Composer/ComposerVC.swift]: @mdhanif-simformsolutions  Please use `LocationManager` class here

File: Sources/StreamChatUI/Composer/ComposerVC.swift [hunk 4]
```diff
@@ -995,6 +1025,67 @@ open class ComposerVC: _ViewController,
         NotificationCenter.default.post(name: .createNewPoll, object: nil, userInfo: userInfo)
     }
 
+    @objc func sendWeatherMessage(_ notification: Notification ) {
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Composer/ComposerVC.swift]: @mdhanif-simformsolutions  lint warning

File: Sources/StreamChatUI/Composer/ComposerVC.swift [hunk 5]
```diff
@@ -995,6 +1025,67 @@ open class ComposerVC: _ViewController,
...
+        let action = AttachmentAction(name: "action", value: "edit", style: .default, type: .button, text: "Edit")
+        ChatClient.shared.messageController(cid: cid, messageId: messageId)
+            .dispatchEphemeralMessageAction(action, getWeatherRequest(notification), key: "weather")
+    }
+
+    func getWeatherRequest(_ notification: Notification) -> [String: RawJSON] {
+        var weatherData = [String: RawJSON]()
+        // Get data from notification
+        let temp = notification.userInfo?["tempreature"] as? String ?? ""
+        let location = notification.userInfo?["location"] as? String ?? ""
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Composer/ComposerVC.swift]: @mdhanif-simformsolutions  get userInfo in separate variable and use it instead.

File: Sources/StreamChatUI/Composer/ComposerVC.swift [hunk 6]
```diff
@@ -995,6 +1025,67 @@ open class ComposerVC: _ViewController,
...
+            var weatherData: [String: Any] = [
+                "latitude": currentLocation.coordinate.latitude,
+                "longitude": currentLocation.coordinate.longitude,
+                "locationName": placemark?.first?.locality]
+            NotificationCenter.default.post(name: .fetchWeather, object: nil, userInfo: weatherData)
+        }
+    }
+
+    private func showSettingsAlert() {
+        let alert = UIAlertController(title: "Permission missing", message: "Please allow location permission from settings to access this feature.", preferredStyle: .alert)
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Composer/ComposerVC.swift]: @mdhanif-simformsolutions  Location Manager class already has these Alert. Please use it and remove duplications

File: Sources/StreamChatUI/Model/WeatherDetailModel.swift [hunk 7]
```diff
@@ -0,0 +1,19 @@
+//
+//  WeatherDetailModel.swift
+//  StreamChatUI
+//
+//  Created by Mohammed Hanif on 10/06/22.
+//  Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+public struct WeatherDetailModel {
+    public var baseUrl = "https://res.cloudinary.com/timeless/image/upload/v1/app/Wallet/Weather/"
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Model/WeatherDetailModel.swift]: @mdhanif-simformsolutions  Please move weather URL in constant file

File: Sources/StreamChatUI/Utils/LocationHelper.swift [hunk 8]
```diff
@@ -0,0 +1,22 @@
+//
+//  LocationHelper.swift
+//  StreamChatUI
+//
+//  Created by Mohammed Hanif on 20/06/22.
+//  Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Utils/LocationHelper.swift]: @mdhanif-simformsolutions  Please remove foundation

File: Sources/StreamChatUI/Utils/LocationHelper.swift [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+import CoreLocation
+
+public class LocationHelper {
+
+    public static let shared = LocationHelper()
+
+    public func getLocationInfo(currentLocation: CLLocation, completion: @escaping CLGeocodeCompletionHandler) {
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Utils/LocationHelper.swift]: @mdhanif-simformsolutions  Shared class not required here.

File: Sources/StreamChatUI/Utils/WeatherHelper.swift [hunk 10]
```diff
@@ -0,0 +1,223 @@
+//
+//  WeatherHelper.swift
+//  StreamChatUI
+//
+//  Created by Mohammed Hanif on 10/06/22.
+//  Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import UIKit
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Utils/WeatherHelper.swift:9]: @mdhanif-simformsolutions  Please remove foundation

File: Sources/StreamChatUI/Utils/WeatherHelper.swift [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+    case thunderstormNight
+
+    // Tornado
+    case tornadoDay
+    case tornadoNight
+}
+
+public class WeatherHelper {
+
+    public static let shared = WeatherHelper()
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Utils/WeatherHelper.swift]: @mdhanif-simformsolutions  Shared instance not required here. Please remove it.

**Pull request merged**